influxql/ast.go

```diff
@@ -642,6 +642,35 @@ type SelectStatement struct {
 	FillValue interface{}
 }
 
+// HasDerivative returns true if one of the fields in the statement is a
+// derivative aggregate
+func (s *SelectStatement) HasDerivative() bool {
+	for _, f := range s.FunctionCalls() {
+		if strings.HasSuffix(f.Name, "derivative") {
+			return true
+		}
+	}
+	return false
+}
+
+// IsSimpleDerivative return true if a field is a derivative function with a
+// variable ref as the first arg
+func (s *SelectStatement) IsSimpleDerivative() bool {
+	for _, f := range s.Fields {
+		if strings.HasSuffix(f.Name(), "derivative") {
```

Comment:
I believe the line below also ensures that it's a `*Call` so it works as is.  In any event, I'll switch it to iterate of `FunctionCalls` to simplify it a bit.